Spec/TestUtilities.swift

```diff
@@ -78,35 +78,46 @@ class AblyTests {
         }
     }
 
-    class func setupOptions(options: ARTClientOptions, debug: Bool = false) -> ARTClientOptions {
-        let request = NSMutableURLRequest(URL: NSURL(string: "https://\(options.restHost):\(options.tlsPort)/apps")!)
-        request.HTTPMethod = "POST"
-        request.HTTPBody = try? appSetupJson["post_apps"].rawData()
-
-        request.allHTTPHeaderFields = [
-            "Accept" : "application/json",
-            "Content-Type" : "application/json"
-        ]
-
-        let (responseData, responseError, _) = NSURLSessionServerTrustSync().get(request)
-
-        if let error = responseError {
-            XCTFail(error.localizedDescription)
-        } else if let data = responseData {
-            let response = JSON(data: data)
+    static var testApplication: JSON?
+    static private var setupOptionsCounter = 0
+
+    class func setupOptions(options: ARTClientOptions, forceNewApp: Bool = false, debug: Bool = false) -> ARTClientOptions {
+        ARTChannels_getChannelNamePrefix = { "test-\(setupOptionsCounter)" }
+        setupOptionsCounter += 1
+
+        if forceNewApp {
+            testApplication = nil
+        }
+
+        guard let app = testApplication else {
+            let request = NSMutableURLRequest(URL: NSURL(string: "https://\(options.restHost):\(options.tlsPort)/apps")!)
+            request.HTTPMethod = "POST"
+            request.HTTPBody = try? appSetupJson["post_apps"].rawData()
+
+            request.allHTTPHeaderFields = [
+                "Accept" : "application/json",
+                "Content-Type" : "application/json"
+            ]
+
+            let (responseData, responseError, _) = NSURLSessionServerTrustSync().get(request)
+
+            if let error = responseError {
+                XCTFail(error.localizedDescription)
+                return options
+            }
+
+            testApplication = JSON(data: responseData!)
             
             if debug {
                 options.logLevel = .Verbose
-                print(response)
+                print(testApplication!)
```

Comment:
Well, it is printing exactly the same as it was doing before. Note the `if debug {`.